apps/src/craft/craft.js

```diff
@@ -161,6 +176,42 @@ Craft.init = function (config) {
   Craft.level = config.level;
   Craft.skin = config.skin;
 
+  var levelTracks = [];
+  if (Craft.level.songs && musicMetadata) {
+    levelTracks = musicMetadata.filter(function(trackMetadata) {
+      return Craft.level.songs.indexOf(trackMetadata.name) !== -1;
+    });
+  }
+
+  Craft.musicController = new MusicController(
+      studioApp.cdoSounds,
+      (filename) => config.skin.assetUrl(`music/${filename}`),
+      levelTracks,
+      levelTracks.length > 1 ? 7500 : null
+  );
+  if (studioApp.cdoSounds && !studioApp.cdoSounds.isAudioUnlocked()) {
+    // Would use addClickTouchEvent, but iOS9 does not let you unlock audio
+    // on touchstart, only on touchend.
+    var removeEvent = dom.addMouseUpTouchEvent(document, function () {
+      studioApp.cdoSounds.unlockAudio();
+      removeEvent();
+    });
+  }
+
+  // Play music when the instructions are shown
+  var playOnce = function () {
+    if (studioApp.cdoSounds && studioApp.cdoSounds.isAudioUnlocked()) {
+      $(window).off('run_button_pressed', playOnce);
+
+      window.setTimeout(() => {
+        var hasSongInlevel = Craft.level.songs && Craft.level.songs.length > 1;
+        var songToPlayFirst = hasSongInlevel ? Craft.level.songs[0] : null;
+        Craft.musicController.play(songToPlayFirst);
+      }, Craft.level.songDelay || 1500);
```

Comment:
:+1: Nicely done - so you enforce the first song, then let it pick randomly after that, right?